mergify_engine/mergify_pull.py

```diff
@@ -342,3 +337,85 @@ def __str__(self):
                 else (self.g_pull.mergeable_state or "none")
             ),
         }
+
+    @functools_bp.cached_property
+    def reviews(self):
+        return list(self.g_pull.get_reviews())
+
+    @functools_bp.cached_property
+    def commits(self):
+        return self.g_pull.get_commits()
+
+    # NOTE(sileht): map all attributes that in theory doesn't do http calls
+
+    @property
+    def number(self):
+        return self.g_pull.number
+
+    @property
+    def title(self):
+        return self.g_pull.title
+
+    @property
+    def user(self):
+        return self.g_pull.user.login
+
+    @property
+    def state(self):
+        return self.g_pull.state
+
+    @property
+    def come_from_same_repo(self):
+        return self.g_pull.head.repo.id == self.g_pull.base.repo.id
+
+    @property
+    def base_is_modifiable(self):
+        return self.g_pull.raw_data["maintainer_can_modify"] or self.come_from_same_repo
+
+    @property
+    def merge_commit_sha(self):
+        return self.g_pull.merge_commit_sha
+
+    @property
+    def head_sha(self):
+        return self.g_pull.head.sha
+
+    @property
+    def base_ref(self):
+        return self.g_pull.base.ref
+
+    @property
+    def head_ref(self):
+        return self.g_pull.head.ref
+
+    @property
+    def base_repo_name(self):
+        return self.g_pull.base.repo.name
+
+    @property
+    def head_repo_name(self):
+        return self.g_pull.head.repo.name
+
+    @property
+    def base_repo_owner(self):
+        return self.g_pull.base.repo.owner.login
+
+    @property
+    def head_repo_owner(self):
+        return self.g_pull.head.repo.owner.login
+
+    @property
+    def mergeable_state(self):
+        return self.g_pull.mergeable_state
+
+    @property
+    def merged(self):
+        return self.g_pull.merged
+
+    @property
+    def merged_by(self):
+        return self.g_pull.merged_by
+
+    @property
+    def stat_commits(self):
```

Comment:
why `stat_commits`?
Is this the number of commits? Should be `num_commits` then?

Reply:
This is not even needed.